samples/constitutions/default/actions.js

```
@@ -1096,6 +1096,39 @@ const actions = new Map([
       },
     ),
   ],
+  [
+    "add_snp_tcb_version",
+    new Action(
+      function (args) {
+        checkType(args.cpuid, "string", "cpuid");
+        checkLength(ccf.strToBuf(args.cpuid), 8, 8, "cpuid");
+        checkLength(hexStrToBuf(args.cpuid), 4, 4, "cpuid");
+
+        checkType(args.tcb_version, "object", "tcb_version");
+        checkType(
+          args.tcb_version?.boot_loader,
+          "number",
+          "tcb_version.boot_loader",
+        );
+        checkType(args.tcb_version?.tee, "number", "tcb_version.tee");
+        checkType(args.tcb_version?.snp, "number", "tcb_version.snp");
+        checkType(
+          args.tcb_version?.microcode,
+          "number",
+          "tcb_version.microcode",
+        );
+      },
+      function (args, proposalId) {
+        // ensure cpuid is uppercase to prevent aliasing
+        ccf.kv["public:ccf.gov.nodes.snp.tcb_versions"].set(
+          ccf.strToBuf(args.cpuid.toUpperCase()),
```

Let's move this to validation, rather than silently convert.